fix: Service timeframe update

cmaddox5 opened the pull request:
#### Summary of changes
**Asana Ticket:** [🏹 Adjust HASTUS service import logic](https://app.asana.com/0/584764604969369/1209777154957390/f)

The original implementation was using the start and end date from `all_calendar.txt` as the start and end date for the `service_date` row. This is not quite right and would cause issues when we integrate with gtfs-creator. The `service_date` rows should only include dates that the service is actually active. For a great write-up on the desired behavior, see this [Miro board](https://miro.com/app/board/uXjVIL8KITs=/).

#### Reviewer Checklist
- [ ] Meets ticket's acceptance criteria
- [ ] Any new or changed functions have typespecs
- [ ] Tests were added for any new functionality (don't just rely on Codecov)
- [ ] This branch was deployed to the staging environment and is currently running with no unexpected increase in warnings, and no errors or crashes.

lemald commented on lib/arrow/hastus/export_upload.ex:
```diff
@@ -164,61 +164,85 @@ defmodule Arrow.Hastus.ExportUpload do
+              &chunk_dates/1
+            )
+
+          %{name: service_id, service_dates: dates ++ additions}
```
**question:** as it stands, it's still possible to have an addition that falls adjacent to one of the other date ranges, right? I'm wondering if that means that it's possible to have examples that would violate the "minimum number of ranges" acceptance criterion.

cmaddox5 replied:
Dang, you're right. Adjacent date ranges would still exist separately in the old logic. Added some additional logic in https://github.com/mbta/arrow/pull/1172/commits/6fd5f3103edc91df359bcb16a0171bc969e0202d that will apply additions to existing timeframes and then merge adjacent timeframes before returning.

cmaddox5 replied:
This involved an additional change to the fixture. The changes are adding additions to `RTL12025-hmb15wg1-Weekday-01` on a Saturday and Sunday to make sure they are added to an existing timeframe. The result is two adjacent timeframes that allows for testing of the merge logic.

cmaddox5 commented:
> **question:** could you give the quick version of what changed in `example.zip` just so I can better understand the testing changes?

[example.zip](Here) is the previous version just in case your curious. Tl;dr, I extended service `RTL12025-hmb15wg1-Weekday-01` so it includes 3/21-3/28, removed `RTL12025-hmb15mo1-Weekday-01` because it is already included in that range, added an exception for service `RTL12025-hmb15wg1-Weekday-01` on 3/26 in `all_calendar_dates.txt`. The rest of the files were unchanged. The original service dates were single days so they weren't really sufficient for testing these changes.

lemald commented on lib/arrow/hastus/export_upload.ex:
```diff
@@ -164,61 +164,134 @@ defmodule Arrow.Hastus.ExportUpload do
+        update_in(acc, [Access.at(i), :end_date], fn _ -> date end)
 
-    imported_service
+      i = Enum.find_index(acc, &(Date.add(&1.start_date, -11) == date)) ->
```
**question:** should this be -1 instead of -11?

cmaddox5 replied:
Yes, thank you.

cmaddox5 replied:
```suggestion
      i = Enum.find_index(acc, &(Date.add(&1.start_date, -1) == date)) ->
```

lemald reviewed (approved):
Code looks good to me and I did some basic local testing which also looked good. 👍
